cmd/server/server.go

```diff
@@ -191,14 +192,18 @@ func (s *server) startService() common.Daemon {
 			params.BlobstoreClient, err = filestore.NewClient(s.cfg.Archival.Filestore)
 		}
 		if s.cfg.Archival.S3store != nil {
-			s3cli, err := s3store.ClientFromConfig(s.cfg.Archival.S3store)
-			if err != nil {
+			var s3cli s3iface.S3API
+			s3cli, err = s3store.ClientFromConfig(s.cfg.Archival.S3store)
```

Comment:
nit: `s3cli, err :=`, do need the var

Reply:
Would that not shadow the err variable, making the later error check never run?
Might have misunderstood you comment, but this is what I'm basing my change on:
https://play.golang.org/p/QvMipO3_ov7

Reply:
How about instead of worrying about this shadowing simply log.Fatal whenever you get an error, it also lets you have better error messages.

```
params.BlobstoreClient, err := filestore.NewClient(s.cfg.Archival.Filestore)
if err != nil {
    log.Fatal("failed to create file based blobstore" ...)
}
...
```